src/unitxt/operators.py

```diff
@@ -179,6 +180,38 @@ def process(
         return flatten_dict(instance, parent_key=self.parent_key, sep=self.sep)
 
 
+class WholeInputFormatter(StreamInstanceOperator):
+    """Generates the whole input to the model, from constant strings that are given as args, and from values found in specified fields of the instance.
```

Comment:
the general_format and input_output_format will be given as args. 
`demos`,` instruction`, `system_prompt` `source` and `target` will be in the instance

Reply:
I interpreted `system_prompt` as `prefix` and `suffix` of ICLformat, and `input_output_format` as all those small pieces. I tried to combine some, but the flow of the code in ICLformat does not allow to reduce the **number** of formatting args.